rackcity/views/pdu_views.py

```diff
@@ -51,6 +55,110 @@ def power_status(request, id):
     )
 
 
+"""
+TODO check that power is in opposite state when performing a toggle
+TODO validate if ports exist/are connected
+"""
+@api_view(['POST'])
+@permission_classes([IsAdminUser])
+def power_on(request):
+    """
+    Turn on power to specified port
+    """
+    data = JSONParser().parse(request)
+    failure_message = ""
+    try:
+        asset = Asset.objects.get(id=data['id'])
+    except Asset.DoesNotExist:
+        failure_message = "No model exists with id=" + str(id)
+        return JsonResponse(
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    power_connections = serialize_power_connections(asset)
+    # Check power is off
+    html = requests.get(pdu_url + get_pdu + get_pdu_status_ext(asset, str(power_connections[data['power_port_number']]['left_right'])))
+    power_status = regex_power_status(html.text, power_connections[data['power_port_number']]['port_number'])[0]
+    if power_status == "ON":
+        return JsonResponse(
+            {"data": "Port is already on"},
+            status=HTTPStatus.OK
+        )
+    toggle_power(asset, data['power_port_number'], "on")
+    return HttpResponse(status=HTTPStatus.OK)
```

Comment:
do we want to send a success_message to the front end to display? or will the view in the front end automatically update?